app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -677,14 +677,32 @@ def up_to_date?(environment = nil, content_view = nil)
       end
 
       def rhsm_url
-        if (rhsm_url_setting = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url').presence)
-          URI(rhsm_url_setting)
-        else
-          # TODO: get this from routes
-          uri = URI.parse(Setting[:foreman_url])
-          uri.path = '/rhsm'
-          uri
-        end
+        rhsm_url_setting = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url').presence
+
+        url = if rhsm_url_setting
+                URI(rhsm_url_setting)
+              else
+                # TODO: get this from routes
+                uri = URI.parse(Setting[:foreman_url])
+                uri.path = '/rhsm'
+                uri
+              end
+
+        # RHSM requires HTTPS for certificate-based authentication
+        unless url.scheme == 'https'
+          error_msg = if rhsm_url_setting
+                        _("Smart Proxy's 'rhsm_url' setting must use HTTPS protocol. " \
+                          "Current value: '%{url}'. Please update the Smart Proxy's rhsm_url setting.") % { url: url.to_s }
```

Comment:
Perhaps it helps to give the Smart Proxy name in the error too because you may not know which one is used and the RHSM URL may be different from the Smart Proxy (for example, in case of a load balanced setup).